### PR TITLE
sxhkd: stop scope before creating

### PR DESCRIPTION
**Pull request by 0qq**
### Description
This allows relogging into graphical session.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/services/sxhkd.nix [hunk 1]
```diff
@@ -73,6 +73,7 @@ in
     ];
 
     xsession.initExtra = ''
+      systemctl --user stop sxhkd.scope
```
Comment (berbiche) [inline modules/services/sxhkd.nix:76]:
```suggestion
      systemctl --user stop sxhkd.scope || true
```

If an error occurs, is the script evaluation aborted?

Comment (0qq) [reply modules/services/sxhkd.nix]: No, just throw an error to journal. I added check if scope is active.

File: modules/services/sxhkd.nix [hunk 2]
```diff
@@ -73,6 +73,8 @@ in
     ];
 
     xsession.initExtra = ''
+      [[ $(systemctl is-active --user sxhkd.scope) == active ]] \
+        && systemctl --user stop sxhkd.scope
```
Comment (ncfavier) [inline modules/services/sxhkd.nix:77]:
```suggestion
      systemctl --user stop sxhkd.scope 2> /dev/null || true
```

This runs into a race condition where the scope is stopped between the two `systemctl` calls.

Don't check, handle failure.

**Review by ncfavier — CHANGES_REQUESTED** [review]

**Review by ncfavier — APPROVED** [review]

**Review by berbiche — APPROVED** [review]
Thanks!

**Pull request merged**